src/main/java/com/mraof/minestuck/client/gui/DialogueScreen.java

```diff
@@ -2,78 +2,148 @@
 
 import com.mojang.blaze3d.systems.RenderSystem;
 import com.mraof.minestuck.entity.dialogue.Dialogue;
+import com.mraof.minestuck.entity.dialogue.DialogueAnimationData;
 import com.mraof.minestuck.network.DialoguePackets;
 import com.mraof.minestuck.network.MSPacketHandler;
 import net.minecraft.client.gui.GuiGraphics;
-import net.minecraft.client.gui.components.Button;
 import net.minecraft.client.gui.components.Tooltip;
 import net.minecraft.client.gui.screens.Screen;
 import net.minecraft.network.chat.Component;
 import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.resources.ResourceLocation;
 import net.minecraft.util.FormattedCharSequence;
-import net.minecraftforge.client.gui.widget.ExtendedButton;
+import software.bernie.geckolib.cache.texture.AnimatableTexture;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 
 /**
- * Displays a screen when interacting with any dialogue capable entity that has a valid dialogue.
- * It is configurable in terms of what the gui image is, what sort of animated talk sprite shows up, and how many response options there are.
+ * This is a GUI which pops up when talking with a dialogue compatible entity.
+ * The corresponding dialogue json file controls the configurable gui texture, npc sprite texture, message, and responses that show up.
  */
 public class DialogueScreen extends Screen
 {
 	private static final int GUI_WIDTH = 224;
 	private static final int GUI_HEIGHT = 176;
 	
+	private static final int BUTTON_GAP = 2;
+	
 	private final int dialogueId;
 	private final Dialogue.DialogueData dialogueData;
+	private final DialogueAnimationData animationData;
 	
 	private int xOffset;
 	private int yOffset;
 	
 	private List<FormattedCharSequence> messageLines;
-	private final List<Button> responseButtons = new ArrayList<>();
+	private final List<List<DialogueButton>> responseButtonPages = new ArrayList<>();
+	private int responsePage = 0;
+	
+	private DialogueButton previousButton;
+	private DialogueButton nextButton;
```

Comment:
I imagine that a scrollbar might feel more natural for this gui than page buttons. Is this something you've thought of?

Reply:
I had considered one but havent implemented one before and decided I might accidentally spend too much time trying to get it to work instead of working on the essentially aspects of this PR. If you have any tips/suggestions, I would be happy to try it out!

Reply:
Sounds like a reasonable decision. Let's stick to it, finish this PR without it and figure it out in a followup PR. And maybe leave a todo here until then.